From 0869c955a3065d0f4aa37a236a15055594f29aae Mon Sep 17 00:00:00 2001
From: Arindam Nath <arindam.nath@amd.com>
Date: Thu, 12 May 2016 12:49:28 +0530
Subject: [PATCH] amd/powerplay: handle power management state based on thermal
 requirements

When two 4k@60Hz monitors are used, DAL calculates the minimum
SCLK required to drive the displays being 39588. This clock
value will cause some systems to exceed the PPT/TDP limits
of 16W.

This workaround will limit the highest possible hard min to
resolve thermal trip issues on such systems.

Signed-off-by: Zedplin Law <Zedplin.Law@amd.com>
Signed-off-by: Ramin Ranjbar <rranjbar@amd.com>
---
 drivers/gpu/drm/amd/powerplay/hwmgr/hardwaremanager.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/amd/powerplay/hwmgr/hardwaremanager.c b/drivers/gpu/drm/amd/powerplay/hwmgr/hardwaremanager.c
index 7888dd8..7ff2c97 100644
--- a/drivers/gpu/drm/amd/powerplay/hwmgr/hardwaremanager.c
+++ b/drivers/gpu/drm/amd/powerplay/hwmgr/hardwaremanager.c
@@ -300,6 +300,12 @@ int phm_store_dal_configuration_data(struct pp_hwmgr *hwmgr,
 		return -EINVAL;
 
 	hwmgr->display_config = *display_config;
+
+	if (hwmgr->display_config.min_core_set_clock > 30000) {
+		hwmgr->display_config.min_core_set_clock = 30000;
+		hwmgr->display_config.min_core_set_clock_in_sr = 30000 / 6;
+	}
+
 	/* to do pass other display configuration in furture */
 
 	if (hwmgr->hwmgr_func->store_cc6_data)
-- 
1.9.1

